src/stseq.c

```diff
@@ -8,6 +8,82 @@
 #include "bufrlib.h"
 #include "mstabs.h"
 
+/** Size of a character string needed to store an FXY value. */
+static const int FXY_STR_LEN = 7;
+
+/**
+ * Define a comparison between two integers.
+ *
+ * This routine defines a comparison between two integers
+ * for use by the binary search function bsearch.
+ *
+ * @param pf1 - first integer to be compared.
+ * @param pf2 - second integer to be compared.
+ *
+ * @return
+ * - -1 PF1 is less than PF2
+ * - 0 PF1 is equal to PF2
+ * - 1 PF1 is greater than PF2
+ *
+ * @author Ator @date 2009-03-23
+ */
+int
+cmpia(const void *pf1, const void *pf2)
+{
+        int *mypf1 = ( int * ) pf1;
+        int *mypf2 = ( int * ) pf2;
+
+        if ( *mypf1 == *mypf2 ) return 0;
+
+        return ( *mypf1 < *mypf2 ? -1 : 1 );
+}
+
+/**
+ * Search for an entry in the BUFR master table.
+ *
+ * This routine searches for an entry corresponding to idn in the bufr
+ * master table (either 'B' or 'D', depending on the value of idn).
+ * The search uses binary search logic, so all of the entries in the
+ * table must be sorted in ascending order (by FXY number) in order
+ * for this routine to work properly.
+ *
+ * @param idn - bit-wise representation of FXY number to be searched for.
+ * @param tab - table in which idn was found ('B' or 'D').
+ * @param ipt - index of entry for idn in master table tab.
+ *
+ * @author J Ator @date 2009-03-23
+*/
+void
+nummtb(int *idn, char *tab, int *ipt)
+{
+        int *pifxyn, *pbs,  nmt;
+
+        char adn[FXY_STR_LEN], errstr[129];
+
+        if ( *idn >= ifxy( "300000", 6 ) ) {
```

Comment:
What's happening here? What is "3000000"? Is that an FXY that is too large? A comment would be helpful and the string "3000000" should be defined as a pre-processor constant instead of being a naked constant in the code.

Reply:
No, this is a test to determine whether it's a Table D descriptor (whose FXY values always have a "3" as their first digit) or a Table B descriptor (whose FXY values always have a "0" as their first digit).  IMHO, the following lines `tab = 'D'` and `tab = 'B'` already clarify this, but if you want I can add a few more comments.  Either way, I don't think this is really a "naked constant" in the sense that you're thinking.